cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperator.java

```diff
@@ -323,7 +324,7 @@ public static int getPodGeneration(Pod resource) {
             long pollingIntervalMs = 1_000;
             long timeoutMs = operationTimeoutMs;
 
-            operation().inNamespace(namespace).withName(name).cascading(cascading).withGracePeriod(-1L).delete();
+            operation().inNamespace(namespace).withName(name).withPropagationPolicy(cascading ? DeletionPropagation.FOREGROUND : DeletionPropagation.BACKGROUND).delete();
```

Comment:
Have you double checked that the gracePeriod is still maintained from the pod when the `withGracePeriod` is removed?